### PR TITLE
fix: cancelling status for SWAP/LIMIT

### LINKED ISSUE #3018: No 'cancelling' state for Swap/Limit orders
1. Place a swap/limit order
2. Run cancel transaction (offchain)

**AR**: once a TX is signed, an order is immediately moves into 'Cancelled' state

**ER**: the order is 'cancelling' state until it is finally cancelled on Explorer

### PR DESCRIPTION
**Pull request by alfetopito**
# Summary

Fix #3018 

# To Test

1. Place a SWAP
2. Cancel it
* Should show `cancelling` for 1 min
3. Repeat with LIMIT
* Should show `cancelling` for 1 min
4. Repeat both with off-chain and on-chain cancellations
* Both should work as expected

### CODE REVIEW

File: apps/cowswap-frontend/src/legacy/state/orders/utils.ts [hunk 1]
```diff
@@ -51,9 +51,7 @@ export function isOrderFulfilled(
  */
 export function isOrderCancelled(order: Pick<EnrichedOrder, 'creationDate' | 'invalidated' | 'status'>): boolean {
   const creationTime = new Date(order.creationDate).getTime()
-  return (
-    (order.invalidated && Date.now() - creationTime > PENDING_ORDERS_BUFFER) || order.status === OrderStatus.CANCELLED
-  )
+  return order.invalidated && Date.now() - creationTime > PENDING_ORDERS_BUFFER
```
Comment (shoom3301) [inline apps/cowswap-frontend/src/legacy/state/orders/utils.ts:54]: This was a part of the fix: https://github.com/cowprotocol/cowswap/pull/2914

Wouldn't it brake it?

Comment (alfetopito) [reply apps/cowswap-frontend/src/legacy/state/orders/utils.ts:54]: Well, to be fair, it's broken already.

This is on barn:
image

But I'll isolate the behaviour just to not block this fix

**Review by elena-zh — COMMENTED** [review]
Hey @alfetopito , great job!

The only one issue I faced is that there is a double 'cancelling' state appears when cancel a market/limit order onchain: 
1. When a TX is executed, I get 'Cancelled' status and notification
2. When it turns into 'Cancelling' again
3. And then I get the 2nd 'Cancelled' status and notification

Could you please take a look at it? Or I should open a separate issue for this?
Thanks!

**Comment by alfetopito** [comment]
> Hey @alfetopito , great job!
> 
> The only one issue I faced is that there is a double 'cancelling' state appears when cancel a market/limit order onchain:
> 
>     1. When a TX is executed, I get 'Cancelled' status and notification
> 
>     2. When it turns into 'Cancelling' again
> 
>     3. And then I get the 2nd 'Cancelled' status and notification
> 
> 
> Could you please take a look at it? Or I should open a separate issue for this? Thanks!



> Hey @alfetopito , great job!
> 
> The only one issue I faced is that there is a double 'cancelling' state appears when cancel a market/limit order onchain:
> 
>     1. When a TX is executed, I get 'Cancelled' status and notification
> 
>     2. When it turns into 'Cancelling' again
> 
>     3. And then I get the 2nd 'Cancelled' status and notification
> 
> 
> Could you please take a look at it? Or I should open a separate issue for this? Thanks!

Thanks @elena-zh !

Should be fixed now

File: apps/cowswap-frontend/src/legacy/state/orders/reducer.ts [hunk 2]
```diff
@@ -314,8 +316,8 @@ export default createReducer(initialState, (builder) =>
               creationTime: newOrder.creationTime,
               composableCowInfo: newOrder.composableCowInfo,
               apiAdditionalInfo: newOrder.apiAdditionalInfo,
-              isCancelling: newOrder.isCancelling,
-              class: newOrder.class,
+              isCancelling: isCancelling,
+              class: orderObj.order.class || newOrder.class, // should never replace existing order class
```
Comment (alfetopito) [inline apps/cowswap-frontend/src/legacy/state/orders/reducer.ts:320]: Additional fix as I noticed class changing from `market` to `limit`.

**Review by elena-zh — APPROVED** [review]
hey @alfetopito , great job!
Looks like the last issue I mentioned was reported in #2325 , so you fixed it.
I still can face #1709 issue while cancelling an order, but it should be addressed separately

**Review by anxolin — COMMENTED** [review]
Approve!

**Pull request merged**